homeassistant/components/miele/sensor.py

```diff
@@ -779,11 +873,15 @@ async def async_added_to_hass(self) -> None:
 
         # recover last value from cache when adding entity
         last_value = await self.async_get_last_state()
-        if last_value and last_value.state != STATE_UNKNOWN:
-            self._last_value = last_value.state
+        if last_value and last_value.state not in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+            self._restore_last_value(last_value.state)
+
+    def _restore_last_value(self, last_value: str) -> None:
+        """Restore the last value from cache."""
+        self._last_value = last_value
```

Comment:
This looks like a bugfix not related to this PR?

Reply:
You're right, sorry. I've dropped a separate PR https://github.com/home-assistant/core/pull/152488